elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,283 @@
+|===
+|Legacy Event | Needs to be added to Elytron
+//|Legacy Location
+
+| Rfc5424SyslogEvent
+| Yes
+//| testsuite/shared/.../syslogserver
+
+| CREATED_ASSERTION
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| RESPONSE_TO_SP
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| ERROR_RESPONSE_TO_SP
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| ERROR_SIG_VALIDATION
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| ERROR_TRUSTED_DOMAIN
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| EXPIRED_ASSERTION
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| LOGIN_INIT
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| LOGIN_COMPLETE
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| REQUEST_FROM_IDP
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| REQUEST_TO_IDP
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| RESPONSE_FROM_IDP
+| Yes
+//| picketlink/.../logging, picketlink/.../metrics, picketlink-federation/picketlink/.../audit
+
+| Invalid Authentication
+| Added Already
+//| picketbox/.../AuditUnitTestCase
+
+| Valid Authentication
+| Added Already
+//| picketbox/.../AuditUnitTestCase
+
+| Invalid Permission
+| Added Already
+//|
+
+| Valid Permission
+| Added Already
+//|
+
+| Audit Log Enabled
+| Yes
+//| wildfly-core/.../AuditLogToSyslogTestCase
+
+| Audit Log Disabled
+| Yes
+//| wildfly-core/.../AuditLogToSyslogTestCase
+
+|===
+* Additional customization attributes should be added based on community feedback, listed in the following table:
+|===
+|Feature |Available in Legacy Auditing |Support Case if Applicable |Implementation Required
+
+| "hostname" attribute
+| No
+| https://access.redhat.com/support/cases/#/case/02188481[Case 02188481]
+| Currently supported in Elytron Audit Logging resulting in just needing the attribute to be added to the management configuration
+
+| Less reliability for increased speed
+| Yes?
+| N/A
+| Ability to configure number of reconnect attempts to a syslog-server. Should be available as an attribute under ```syslogAuditLogType``` in the Elytron schema
+
+|===
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Test Plan
+
+Tests for the new Audit Logging capabilities will be added.
```

Comment:
Which tests and where they should be added?

Reply:
The tests will be added in the same package as where the code is added and will verify that the added Endpoints/formatters/etc work correctly, I will put additional detail for this in the next update